Loading environment in migration tasks

ProGM opened the pull request:
Temporary fix for #1210 

Probably it would not work for non-rails, but for now it's a good solution, I think.

This pull introduces/changes:
 * Initialized environment in all migration rake tasks



Pings:
@cheerfulstoic
@subvertallchris

cheerfulstoic commented:
This seems fine.  I think maybe we depend on `environment` elsewhere, but for non-Rails apps we may need to define an `environment` task which can be overridden by Rails (or whatever app-specific stuff) so that it doesn't just fail.  If you've checked that this works in non-Rails I'm happy to have the change

ProGM commented:
@cheerfulstoic Uhm, you're right, maybe I should define a custom environment task to be overwritten by rails

ProGM commented:
@cheerfulstoic I rebased master to avoid conflicts with #1235 

I moved the part that creates the neo4j session in a new class `Neo4j::SessionManager` and I conditionally defined a `environment` task, that simply creates the session.

Now migrations should work both inside and outside rails ;)
